### ISSUE TITLE
Azure Cloud template parsing or snapshot generation logic need an update to support dependency between resources.

### ISSUE THREAD

**Issue by rezoan** [prancer-io/cloud-validation-framework#485]
Please follow the point 3 mentioned in issue: https://github.com/prancer-io/cloud-validation-framework/issues/416

**Comment by rezoan** [comment]
```
an example live snapshot:

{
  "resource_group": "dev-rezoan",
  "resources": [
    {
      "apiVersion": "2021-08-01-preview",
      "kind": "v12.0",
      "location": "eastus",
      "name": "r4redudbserver",
      "properties": {
        "administratorLogin": "r4redu",
        "minimalTlsVersion": "1.1",
        "publicNetworkAccess": "Enabled",
        "restrictOutboundNetworkAccess": "Disabled",
        "version": "12.0"
      },
      "type": "Microsoft.Sql/servers"
    },
    {
      "apiVersion": "2014-04-01",
      "dependsOn": [
        "[resourceId('Microsoft.Sql/servers', 'r4redudbserver')]"
      ],
      "location": "East US",
      "name": "r4redudbserver/Default",
      "properties": {
        "auditingState": "Disabled"
      },
      "type": "Microsoft.Sql/servers/auditingPolicies"
    },
    {
      "apiVersion": "2021-08-01-preview",
      "dependsOn": [
        "[resourceId('Microsoft.Sql/servers', 'r4redudbserver')]"
      ],
      "name": "r4redudbserver/Default",
      "properties": {
        "auditActionsAndGroups": [],
        "isAzureMonitorTargetEnabled": false,
        "isStorageSecondaryKeyInUse": false,
        "retentionDays": 0,
        "state": "Disabled",
        "storageAccountSubscriptionId": "00000000-0000-0000-0000-000000000000"
      },
      "type": "Microsoft.Sql/servers/auditingSettings"
    },
    {
      "apiVersion": "2021-08-01-preview",
      "dependsOn": [
        "[resourceId('Microsoft.Sql/servers', 'r4redudbserver')]"
      ],
      "location": "eastus",
      "name": "r4redudbserver/default",
      "properties": {
        "connectionType": "Default"
      },
      "type": "Microsoft.Sql/servers/connectionPolicies"
    },
    {
      "apiVersion": "2021-08-01-preview",
      "dependsOn": [
        "[resourceId('Microsoft.Sql/servers', 'r4redudbserver')]"
      ],
      "kind": "v12.0,user,vcore",
      "location": "eastus",
      "name": "r4redudbserver/r4redudb",
      "properties": {
        "catalogCollation": "SQL_Latin1_General_CP1_CI_AS",
        "collation": "SQL_Latin1_General_CP1_CI_AS",
        "isLedgerOn": false,
        "licenseType": "LicenseIncluded",
        "maintenanceConfigurationId": "/subscriptions/a6941677-4c37-42fb-960c-dad8f25060a3/providers/Microsoft.Maintenance/publicMaintenanceConfigurations/SQL_Default",
        "maxSizeBytes": 34359738368,
        "readScale": "Disabled",
        "requestedBackupStorageRedundancy": "Geo",
        "zoneRedundant": false
      },
      "sku": {
        "capacity": 2,
        "family": "Gen5",
        "name": "GP_Gen5",
        "tier": "GeneralPurpose"
      },
      "type": "Microsoft.Sql/servers/databases"
    },
    {
      "apiVersion": "2014-04-01",
      "dependsOn": [
        "[resourceId('Microsoft.Sql/servers', 'r4redudbserver')]"
      ],
      "location": "East US",
      "name": "r4redudbserver/master/Default",
      "properties": {
        "auditingState": "Disabled"
      },
      "type": "Microsoft.Sql/servers/databases/auditingPolicies"
    },
    {
      "apiVersion": "2021-08-01-preview",
      "dependsOn": [
        "[resourceId('Microsoft.Sql/servers', 'r4redudbserver')]"
      ],
      "name": "r4redudbserver/master/Default",
      "properties": {
        "isAzureMonitorTargetEnabled": false,
        "retentionDays": 0,
        "state": "Disabled",
        "storageAccountSubscriptionId": "00000000-0000-0000-0000-000000000000"
      },
      "type": "Microsoft.Sql/servers/databases/auditingSettings"
    },
    {
      "apiVersion": "2021-08-01-preview",
      "dependsOn": [
        "[resourceId('Microsoft.Sql/servers', 'r4redudbserver')]"
      ],
      "name": "r4redudbserver/master/Default",
      "properties": {
        "isAzureMonitorTargetEnabled": false,
        "retentionDays": 0,
        "state": "Disabled",
        "storageAccountSubscriptionId": "00000000-0000-0000-0000-000000000000"
      },
      "type": "Microsoft.Sql/servers/databases/extendedAuditingSettings"
    },
    {
      "apiVersion": "2014-04-01",
      "dependsOn": [
        "[resourceId('Microsoft.Sql/servers', 'r4redudbserver')]"
      ],
      "location": "East US",
      "name": "r4redudbserver/master/Default",
      "properties": {
        "state": "Disabled"
      },
      "type": "Microsoft.Sql/servers/databases/geoBackupPolicies"
    },
    {
      "apiVersion": "2021-08-01-preview",
      "dependsOn": [
        "[resourceId('Microsoft.Sql/servers', 'r4redudbserver')]"
      ],
      "name": "r4redudbserver/master/Current",
      "properties": {},
      "type": "Microsoft.Sql/servers/databases/ledgerDigestUploads"
    },
    {
      "apiVersion": "2021-08-01-preview",
      "dependsOn": [
        "[resourceId('Microsoft.Sql/servers', 'r4redudbserver')]"
      ],
      "name": "r4redudbserver/master/Default",
      "properties": {
        "disabledAlerts": [
          ""
        ],
        "emailAccountAdmins": false,
        "emailAddresses": [
          ""
        ],
        "retentionDays": 0,
        "state": "Disabled"
      },
      "type": "Microsoft.Sql/servers/databases/securityAlertPolicies"
    },
    {
      "apiVersion": "2021-08-01-preview",
      "dependsOn": [
        "[resourceId('Microsoft.Sql/servers', 'r4redudbserver')]"
      ],
      "name": "r4redudbserver/master/Current",
      "properties": {
        "state": "Disabled"
      },
      "type": "Microsoft.Sql/servers/databases/transparentDataEncryption"
    },
    {
      "apiVersion": "2021-08-01-preview",
      "dependsOn": [
        "[resourceId('Microsoft.Sql/servers', 'r4redudbserver')]"
      ],
      "name": "r4redudbserver/master/Default",
      "properties": {
        "recurringScans": {
          "emailSubscriptionAdmins": true,
          "isEnabled": false
        }
      },
      "type": "Microsoft.Sql/servers/databases/vulnerabilityAssessments"
    },
    {
      "apiVersion": "2021-08-01-preview",
      "dependsOn": [
        "[resourceId('Microsoft.Sql/servers', 'r4redudbserver')]"
      ],
      "name": "r4redudbserver/Default",
      "properties": {
        "isAzureMonitorTargetEnabled": false,
        "state": "Disabled",
        "storageAccountSubscriptionId": "00000000-0000-0000-0000-000000000000"
      },
      "type": "Microsoft.Sql/servers/devOpsAuditingSettings"
    },
    {
      "apiVersion": "2021-08-01-preview",
      "dependsOn": [
        "[resourceId('Microsoft.Sql/servers', 'r4redudbserver')]"
      ],
      "kind": "servicemanaged",
      "name": "r4redudbserver/current",
      "properties": {
        "autoRotationEnabled": false,
        "serverKeyName": "ServiceManaged",
        "serverKeyType": "ServiceManaged"
      },
      "type": "Microsoft.Sql/servers/encryptionProtector"
    },
    {
      "apiVersion": "2021-08-01-preview",
      "dependsOn": [
        "[resourceId('Microsoft.Sql/servers', 'r4redudbserver')]"
      ],
      "name": "r4redudbserver/Default",
      "properties": {
        "auditActionsAndGroups": [],
        "isAzureMonitorTargetEnabled": false,
        "isStorageSecondaryKeyInUse": false,
        "retentionDays": 0,
        "state": "Disabled",
        "storageAccountSubscriptionId": "00000000-0000-0000-0000-000000000000"
      },
      "type": "Microsoft.Sql/servers/extendedAuditingSettings"
    },
    {
      "apiVersion": "2021-08-01-preview",
      "dependsOn": [
        "[resourceId('Microsoft.Sql/servers', 'r4redudbserver')]"
      ],
      "kind": "servicemanaged",
      "name": "r4redudbserver/ServiceManaged",
      "properties": {
        "serverKeyType": "ServiceManaged"
      },
      "type": "Microsoft.Sql/servers/keys"
    },
    {
      "apiVersion": "2021-08-01-preview",
      "dependsOn": [
        "[resourceId('Microsoft.Sql/servers', 'r4redudbserver')]"
      ],
      "name": "r4redudbserver/Default",
      "properties": {
        "disabledAlerts": [
          ""
        ],
        "emailAccountAdmins": false,
        "emailAddresses": [
          ""
        ],
        "retentionDays": 0,
        "state": "Disabled"
      },
      "type": "Microsoft.Sql/servers/securityAlertPolicies"
    },
    {
      "apiVersion": "2021-08-01-preview",
      "dependsOn": [
        "[resourceId('Microsoft.Sql/servers', 'r4redudbserver')]"
      ],
      "name": "r4redudbserver/Default",
      "properties": {
        "recurringScans": {
          "emailSubscriptionAdmins": true,
          "isEnabled": false
        },
        "storageContainerPath": null
      },
      "type": "Microsoft.Sql/servers/vulnerabilityAssessments"
    },
    {
      "apiVersion": "2014-04-01",
      "dependsOn": [
        "[resourceId('Microsoft.Sql/servers/databases', 'r4redudbserver', 'r4redudb')]",
        "[resourceId('Microsoft.Sql/servers', 'r4redudbserver')]"
      ],
      "name": "r4redudbserver/r4redudb/CreateIndex",
      "properties": {
        "autoExecuteValue": "Disabled"
      },
      "type": "Microsoft.Sql/servers/databases/advisors"
    },
    {
      "apiVersion": "2014-04-01",
      "dependsOn": [
        "[resourceId('Microsoft.Sql/servers/databases', 'r4redudbserver', 'r4redudb')]",
        "[resourceId('Microsoft.Sql/servers', 'r4redudbserver')]"
      ],
      "name": "r4redudbserver/r4redudb/DbParameterization",
      "properties": {
        "autoExecuteValue": "Disabled"
      },
      "type": "Microsoft.Sql/servers/databases/advisors"
    },
    {
      "apiVersion": "2014-04-01",
      "dependsOn": [
        "[resourceId('Microsoft.Sql/servers/databases', 'r4redudbserver', 'r4redudb')]",
        "[resourceId('Microsoft.Sql/servers', 'r4redudbserver')]"
      ],
      "name": "r4redudbserver/r4redudb/DefragmentIndex",
      "properties": {
        "autoExecuteValue": "Disabled"
      },
      "type": "Microsoft.Sql/servers/databases/advisors"
    },
    {
      "apiVersion": "2014-04-01",
      "dependsOn": [
        "[resourceId('Microsoft.Sql/servers/databases', 'r4redudbserver', 'r4redudb')]",
        "[resourceId('Microsoft.Sql/servers', 'r4redudbserver')]"
      ],
      "name": "r4redudbserver/r4redudb/DropIndex",
      "properties": {
        "autoExecuteValue": "Disabled"
      },
      "type": "Microsoft.Sql/servers/databases/advisors"
    },
    {
      "apiVersion": "2014-04-01",
      "dependsOn": [
        "[resourceId('Microsoft.Sql/servers/databases', 'r4redudbserver', 'r4redudb')]",
        "[resourceId('Microsoft.Sql/servers', 'r4redudbserver')]"
      ],
      "name": "r4redudbserver/r4redudb/ForceLastGoodPlan",
      "properties": {
        "autoExecuteValue": "Enabled"
      },
      "type": "Microsoft.Sql/servers/databases/advisors"
    },
    {
      "apiVersion": "2014-04-01",
      "dependsOn": [
        "[resourceId('Microsoft.Sql/servers/databases', 'r4redudbserver', 'r4redudb')]",
        "[resourceId('Microsoft.Sql/servers', 'r4redudbserver')]"
      ],
      "location": "East US",
      "name": "r4redudbserver/r4redudb/Default",
      "properties": {
        "auditingState": "Disabled"
      },
      "type": "Microsoft.Sql/servers/databases/auditingPolicies"
    },
    {
      "apiVersion": "2021-08-01-preview",
      "dependsOn": [
        "[resourceId('Microsoft.Sql/servers/databases', 'r4redudbserver', 'r4redudb')]",
        "[resourceId('Microsoft.Sql/servers', 'r4redudbserver')]"
      ],
      "name": "r4redudbserver/r4redudb/Default",
      "properties": {
        "isAzureMonitorTargetEnabled": false,
        "retentionDays": 0,
        "state": "Disabled",
        "storageAccountSubscriptionId": "00000000-0000-0000-0000-000000000000"
      },
      "type": "Microsoft.Sql/servers/databases/auditingSettings"
    },
    {
      "apiVersion": "2021-08-01-preview",
      "dependsOn": [
        "[resourceId('Microsoft.Sql/servers/databases', 'r4redudbserver', 'r4redudb')]",
        "[resourceId('Microsoft.Sql/servers', 'r4redudbserver')]"
      ],
      "name": "r4redudbserver/r4redudb/default",
      "properties": {
        "monthlyRetention": "PT0S",
        "weekOfYear": 0,
        "weeklyRetention": "PT0S",
        "yearlyRetention": "PT0S"
      },
      "type": "Microsoft.Sql/servers/databases/backupLongTermRetentionPolicies"
    },
    {
      "apiVersion": "2021-08-01-preview",
      "dependsOn": [
        "[resourceId('Microsoft.Sql/servers/databases', 'r4redudbserver', 'r4redudb')]",
        "[resourceId('Microsoft.Sql/servers', 'r4redudbserver')]"
      ],
      "name": "r4redudbserver/r4redudb/default",
      "properties": {
        "diffBackupIntervalInHours": 12,
        "retentionDays": 7
      },
      "type": "Microsoft.Sql/servers/databases/backupShortTermRetentionPolicies"
    },
    {
      "apiVersion": "2021-08-01-preview",
      "dependsOn": [
        "[resourceId('Microsoft.Sql/servers/databases', 'r4redudbserver', 'r4redudb')]",
        "[resourceId('Microsoft.Sql/servers', 'r4redudbserver')]"
      ],
      "name": "r4redudbserver/r4redudb/Default",
      "properties": {
        "isAzureMonitorTargetEnabled": false,
        "retentionDays": 0,
        "state": "Disabled",
        "storageAccountSubscriptionId": "00000000-0000-0000-0000-000000000000"
      },
      "type": "Microsoft.Sql/servers/databases/extendedAuditingSettings"
    },
    {
      "apiVersion": "2014-04-01",
      "dependsOn": [
        "[resourceId('Microsoft.Sql/servers/databases', 'r4redudbserver', 'r4redudb')]",
        "[resourceId('Microsoft.Sql/servers', 'r4redudbserver')]"
      ],
      "location": "East US",
      "name": "r4redudbserver/r4redudb/Default",
      "properties": {
        "state": "Enabled"
      },
      "type": "Microsoft.Sql/servers/databases/geoBackupPolicies"
    },
    {
      "apiVersion": "2021-08-01-preview",
      "dependsOn": [
        "[resourceId('Microsoft.Sql/servers/databases', 'r4redudbserver', 'r4redudb')]",
        "[resourceId('Microsoft.Sql/servers', 'r4redudbserver')]"
      ],
      "name": "r4redudbserver/r4redudb/Current",
      "properties": {},
      "type": "Microsoft.Sql/servers/databases/ledgerDigestUploads"
    },
    {
      "apiVersion": "2021-08-01-preview",
      "dependsOn": [
        "[resourceId('Microsoft.Sql/servers/databases', 'r4redudbserver', 'r4redudb')]",
        "[resourceId('Microsoft.Sql/servers', 'r4redudbserver')]"
      ],
      "name": "r4redudbserver/r4redudb/Default",
      "properties": {
        "disabledAlerts": [
          ""
        ],
        "emailAccountAdmins": false,
        "emailAddresses": [
          ""
        ],
        "retentionDays": 0,
        "state": "Disabled"
      },
      "type": "Microsoft.Sql/servers/databases/securityAlertPolicies"
    },
    {
      "apiVersion": "2021-08-01-preview",
      "dependsOn": [
        "[resourceId('Microsoft.Sql/servers/databases', 'r4redudbserver', 'r4redudb')]",
        "[resourceId('Microsoft.Sql/servers', 'r4redudbserver')]"
      ],
      "name": "r4redudbserver/r4redudb/Current",
      "properties": {
        "state": "Enabled"
      },
      "type": "Microsoft.Sql/servers/databases/transparentDataEncryption"
    },
    {
      "apiVersion": "2021-08-01-preview",
      "dependsOn": [
        "[resourceId('Microsoft.Sql/servers/databases', 'r4redudbserver', 'r4redudb')]",
        "[resourceId('Microsoft.Sql/servers', 'r4redudbserver')]"
      ],
      "name": "r4redudbserver/r4redudb/Default",
      "properties": {
        "recurringScans": {
          "emailSubscriptionAdmins": true,
          "emails": [],
          "isEnabled": false
        }
      },
      "type": "Microsoft.Sql/servers/databases/vulnerabilityAssessments"
    }
  ],
  "subscription_id": "0000000-0000-000-000-000000000"
}
```

**Comment by vatsalgit5118** [comment]
Updated the Azure crawler to process the `resourceId` function in the `dependsOn` field. Released enhancement in version 2.1.8
